### PR TITLE
feat: add Captcha support with CaptchaFox

### PR DESCRIPTION
**Pull request by wpf500**
Add Captcha support for when submitting callout responses

image

### CODE REVIEW

File: src/components/pages/admin/callouts/callouts.interface.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ export interface TitleAndImageStepProps {
 export interface SettingsStepProps {
   whoCanTakePart: 'members' | 'everyone';
   allowAnonymousResponses: 'none' | 'guests' | 'all';
+  requireCaptcha: 'none' | 'guest' | 'all';
```
Comment (JumpLink) [inline src/components/pages/admin/callouts/callouts.interface.ts:42]: Does it make sense to move the enum from the backend to common and use it here? Then it only needs to be used in one place

File: src/utils/api/callout.ts [hunk 2]
```diff
@@ -108,13 +108,18 @@ export async function fetchResponsesForMap(
...
-    guestEmail: data.guestEmail,
-  });
+  await instance.post(
+    `/callout/${slug}/responses`,
+    {
+      answers: data.answers,
+      guestName: data.guestName,
+      guestEmail: data.guestEmail,
+    },
+    { params: { captchaToken } }
```
Comment (JumpLink) [inline src/utils/api/callout.ts:121]: It is no problem to send the parameter even if it is not required?

**Review by JumpLink — APPROVED** [review]
Looks good, just two questions

File: src/utils/api/callout.ts (hunk 2 above)
Comment (wpf500) [reply src/utils/api/callout.ts:121]: If `captchaToken` is undefined it won't actually send the parameter I think, although I haven't actually checked that

File: src/components/pages/admin/callouts/callouts.interface.ts (hunk 1 above)
Comment (wpf500) [reply src/components/pages/admin/callouts/callouts.interface.ts:42]: It does, I was just being lazy :upside_down_face:. I will fix

Comment (wpf500) [reply src/components/pages/admin/callouts/callouts.interface.ts:42]: I will fix this when updating beabee-common to your new version as the current one doesn't have any enums defined in it and seems pointless to add it when it's about to be removed in any case

**Pull request merged**